### PR TITLE
JW-226 Missing DeleteSymlinkDevice

### PR DESCRIPTION
**Pull request by mmasztalerczuk**

### CODE REVIEW

File: windows/vr_ksync.c [hunk 1]
```diff
@@ -165,6 +182,7 @@ CreateDevice(PDRIVER_OBJECT DriverObject)
 
     if (NT_SUCCESS(Status))
     {
+		ToClean |= DEVICE;
```
Comment (Attil) [inline windows/vr_ksync.c]: Tabs/spaces mismatch I think.

File: windows/vr_ksync.c [hunk 2]
```diff
@@ -178,13 +196,10 @@ CreateDevice(PDRIVER_OBJECT DriverObject)
...
 
-        if (NT_WARNING(Status) || NT_INFORMATION(Status))
+        if (NT_WARNING(Status) || NT_INFORMATION(Status) || NT_SUCCESS(Status))
         {
             DbgPrint("IoCreateSymbolicLink %d\n", Status);
-        }
-        else if (NT_ERROR(Status))
-        {
-            DestroyDevice(DriverObject);
+			ToClean |= SYMLINK;
```
Comment (Attil) [inline windows/vr_ksync.c]: Tabs/spaces mismatch I think.

File: windows/vrouter_mod.c [hunk 3]
```diff
@@ -185,6 +185,7 @@ SxExtInitialize(PDRIVER_OBJECT DriverObject)
 VOID
 SxExtUninitialize(PDRIVER_OBJECT DriverObject)
 {
+	DestroyDevice(DriverObject);
```
Comment (Attil) [inline windows/vrouter_mod.c]: Tabs/spaces mismatch I think.

File: windows/vr_ksync.c [hunk 4]
```diff
@@ -134,7 +139,19 @@ NotImplemented(PDEVICE_OBJECT DriverObject, PIRP Irp)
...
+	{
+		RtlUnicodeStringInit(&_DeviceSymLink, DeviceSymLink);
+		IoDeleteSymbolicLink(&_DeviceSymLink);
+		ToClean ^= SYMLINK;
+	}
+	if (ToClean & DEVICE)
+	{
+		IoDeleteDevice(DriverObject->DeviceObject);
+		ToClean ^= DEVICE;
+	}
```
Comment (sodar) [inline windows/vr_ksync.c]: Please use spaces instead of tabs.

File: windows/vr_ksync.c [hunk 5]
```diff
@@ -6,6 +6,11 @@
 const WCHAR DeviceName[] = L"\\Device\\vrouterKsync";
 const WCHAR DeviceSymLink[] = L"\\DosDevices\\vrouterKsync";
 
+#define SYMLINK 1
+#define DEVICE 2
+
+int ToClean = 0;
```
Comment (sodar) [inline windows/vr_ksync.c]: If `ToClean` is used only in `vr_ksync.c`, then maybe it can be marked as static variable?

**Pull request merged**